src/Symfony/Bundle/FrameworkBundle/EventListener/SessionListener.php

```diff
@@ -14,10 +14,14 @@
 use Symfony\Component\HttpKernel\EventListener\SessionListener as BaseSessionListener;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
+@trigger_error(sprintf('The %s class is deprecated since version 3.3 and will be removed in 4.0. Use %s instead.', SessionListener::class, BaseSessionListener::class), E_USER_DEPRECATED);
```

Comment:
I know. I suppose I make this comment here as getter injection is probably the most controversial feature and probably a good candidate for removal. "service locators" seems less controversial to me. But I'm probably biased here. Let's wait for more feedback from @symfony/deciders.